### PR TITLE
Dataiku integration tip

### LINKED ISSUE #830: Add Dataiku integration tip
**Description**
Add an article like the one for Databricks but for Dataiku.

**Expected change**
Create a new tip/article for Dataiku

### PR DESCRIPTION
**Pull request by FlorianJacta**
Fixes: #830 

Unzip this file to see the built tip:

[Uploading Integrating with Dataiku - Taipy (2).zip…]()

### CODE REVIEW

File: docs/knowledge_base/tips/dataiku/index.md
Comment (jrobinAV) [inline docs/knowledge_base/tips/dataiku/index.md]:
```suggestion
title: Integrating with Dataiku
```

File: docs/knowledge_base/tips/dataiku/index.md
Comment (jrobinAV) [inline docs/knowledge_base/tips/dataiku/index.md]:
```suggestion
Integration with external platforms is often essential for executing tasks and functions
efficiently. This article presents a comprehensive guide on integrating Dataiku with Taipy,
a tool for orchestrating tasks and exploring various iterations of business problems. By
integrating Dataiku projects with Taipy, users can enhance their data processing and
visualization capabilities.
```

File: docs/knowledge_base/tips/dataiku/index.md
Comment (jrobinAV) [inline docs/knowledge_base/tips/dataiku/index.md]:
```suggestion
Start by ensuring that you have the required libraries installed. You'll need 
`dataiku-api-client` to access Dataiku and `taipy-gui` to build the GUI. Install these 
libraries using pip:
```

File: docs/knowledge_base/tips/dataiku/index.md
Comment (jrobinAV) [inline docs/knowledge_base/tips/dataiku/index.md]:
```suggestion
Use the `DSSClient` from the `dataikuapi` package to establish a connection with your Dataiku
instance. Replace `"http://your-dataiku-instance.com"` and `"your_api_key"` with your actual
Dataiku host and API key.
```

File: docs/knowledge_base/tips/dataiku/index.md
Comment (jrobinAV) [inline docs/knowledge_base/tips/dataiku/index.md]:
```suggestion
Fetch data from your Dataiku projects by iterating through the projects and collecting
relevant metrics for each model. This example code retrieves metrics like AUC and organizes
them into a `pandas.DataFrame`.
```

File: docs/knowledge_base/tips/dataiku/index.md
Comment (jrobinAV) [inline docs/knowledge_base/tips/dataiku/index.md]:
```suggestion
Before visualization, you may need to prepare your data further. For instance, you can
concatenate certain columns to form a unique identifier or name for each entry. Here, we add
a 'Name' column to our DataFrame.
```

File: docs/knowledge_base/tips/dataiku/index.md
Comment (jrobinAV) [inline docs/knowledge_base/tips/dataiku/index.md]:
```suggestion
Define the GUI using Taipy's markdown-like syntax. This GUI will include dynamic selectors for projects and metrics, a table to display data, and a bar chart for visualizing metric values.
```

File: docs/knowledge_base/tips/dataiku/index.md
Comment (jrobinAV) [inline docs/knowledge_base/tips/dataiku/index.md]:
```suggestion
Creating and executing projects on Dataiku involves several steps, from setting up your
Dataiku instance to defining and running projects. Here's a step-by-step guide on how to
seamlessly integrate Dataiku projects with Taipy scenarios:
```

File: docs/knowledge_base/tips/dataiku/index.md
Comment (jrobinAV) [inline docs/knowledge_base/tips/dataiku/index.md]: In the article, I would assume these steps (1 and 2) as a prerequisite. It is already done in Dataiku.
We don't want to document how to create a Dataiku project. This is not our goal plus it's going to be painful to maintain.

File: docs/knowledge_base/tips/dataiku/index.md
Comment (jrobinAV) [inline docs/knowledge_base/tips/dataiku/index.md]:
```suggestion
By adapting the given Dataiku API script to work with a generic data node in Taipy, you can
define reading and writing functions to interact with Dataiku datasets. Then, configure a
generic data node to use these functions for data handling.
```

File: docs/knowledge_base/tips/dataiku/index.md
Comment (jrobinAV) [inline docs/knowledge_base/tips/dataiku/index.md]: Can you have a title showing the motivation ?

"Run a Dataiku pipeline from your Taipy application"
"Let your end-user run a Dataiku pipeline"
etc.

File: docs/knowledge_base/tips/dataiku/index.md
Comment (jrobinAV) [inline docs/knowledge_base/tips/dataiku/index.md:138]: I would add a small introduction explaining the rough idea, which is encapsulating a Dataiku call to run a pipeline into a taipy Task.

File: docs/knowledge_base/tips/dataiku/index.md
Comment (jrobinAV) [inline docs/knowledge_base/tips/dataiku/index.md]:
```suggestion
With data nodes handling data autonomously, tasks now focus on action triggers, such as
starting Dataiku flows. Let's assume we have a function `trigger_dataiku_flow` that initiates
a specific flow within a Dataiku project:
```

File: docs/knowledge_base/tips/dataiku/index.md
Comment (jrobinAV) [inline docs/knowledge_base/tips/dataiku/index.md]:
```suggestion
This scenario outlines a workflow where Taipy manages the overall process, including the
initiation of Dataiku flows, with data nodes autonomously handling data exchanges with
Dataiku.
```

File: docs/knowledge_base/tips/dataiku/index.md
Comment (jrobinAV) [inline docs/knowledge_base/tips/dataiku/index.md]:
```suggestion
To execute the scenario, use Taipy's execution model. This initiates the configured workflow,
including autonomous data exchange with Dataiku and triggering the specified flow:
```

File: docs/knowledge_base/tips/dataiku/index.md
Comment (jrobinAV) [inline docs/knowledge_base/tips/dataiku/index.md]:
```suggestion
This streamlined approach, where tasks focus on action triggers (like starting Dataiku flows)
while data nodes manage data reading and writing, enhancing the interaction between the 
end-user and its data and models.
```

File: docs/knowledge_base/tips/dataiku/index.md
Comment (jrobinAV) [inline docs/knowledge_base/tips/dataiku/index.md]:
```suggestion
In conclusion, integrating Dataiku with Taipy provides a powerful solution for streamlining
data processing, visualization, and workflow orchestration. By leveraging Taipy's scenario
management and Dataiku's robust capabilities, organizations can enhance efficiency, scalability,
and collaboration in their data workflows. This integration enables seamless interaction between
Dataiku projects and Taipy, facilitating the creation and scalability of data-driven
applications.
```

**Comment by FlorianJacta** [comment]
New update of article:
- New code
- New structure
- Suggestions accepted
- Code available
- Screenshot

[Integrating with Dataiku - Taipy (2).zip](https://github.com/Avaiga/taipy-doc/files/14311040/Integrating.with.Dataiku.-.Taipy.2.zip)

image

image

**Comment by FlorianJacta** [comment]
Waiting @marisogo for illustration

Done!

**Review by jrobinAV — APPROVED** [review]

**Comment by FlorianJacta** [comment]
Waiting for Dataiku approval

**Comment by jrobinAV** [comment]
> Waiting for Dataiku approval

I am adding the label blocked until we can proceed.

**Comment by jrobinAV** [comment]
@FlorianJacta, Any news on that?

**Comment by FlorianJacta** [comment]
Waiting Dataiku answer still

**Comment by jrobinAV** [comment]
@FlorianJacta Can we delete the branch?

**Pull request merged**
